examples/js/loaders/GLTFLoader.js

```diff
@@ -310,31 +304,27 @@ THREE.GLTFLoader = ( function () {
 					lightNode = new THREE.SpotLight( color );
 					lightNode.distance = range;
 					// Handle spotlight properties.
-					light.spot = light.spot || {};
-					light.spot.innerConeAngle = light.spot.innerConeAngle !== undefined ? light.spot.innerConeAngle : 0;
-					light.spot.outerConeAngle = light.spot.outerConeAngle !== undefined ? light.spot.outerConeAngle : Math.PI / 4.0;
-					lightNode.angle = light.spot.outerConeAngle;
-					lightNode.penumbra = 1.0 - light.spot.innerConeAngle / light.spot.outerConeAngle;
+					lightDef.spot = lightDef.spot || {};
+					lightDef.spot.innerConeAngle = lightDef.spot.innerConeAngle !== undefined ? lightDef.spot.innerConeAngle : 0;
+					lightDef.spot.outerConeAngle = lightDef.spot.outerConeAngle !== undefined ? lightDef.spot.outerConeAngle : Math.PI / 4.0;
+					lightNode.angle = lightDef.spot.outerConeAngle;
+					lightNode.penumbra = 1.0 - lightDef.spot.innerConeAngle / lightDef.spot.outerConeAngle;
 					lightNode.target.position.set( 0, 0, 1 );
 					lightNode.add( lightNode.target );
 					break;
 
-			}
-
-			if ( lightNode ) {
-
-				lightNode.decay = 2;
+				default:
+					throw new Error( 'THREE.GLTFLoader: Unexpected light type, "' + lightDef.type + '".' );
 
-				if ( light.intensity !== undefined ) {
+			}
 
-					lightNode.intensity = light.intensity;
+			lightNode.decay = 2;
 
-				}
+			if ( lightDef.intensity !== undefined ) lightNode.intensity = lightDef.intensity;
 
-				lightNode.name = light.name || ( 'light_' + lightId );
-				this.lights[ lightId ] = lightNode;
+			lightNode.name = lightDef.name || ( 'light_' + i );
```

Comment:
This isn't new in this PR but I think the following would be better

    lightNode.name = lightDef.name !== undefined ? lightDef.name : ( 'light_' + i );

because `lightDef.name` can be empty strings `''`.

Reply:
BTW any special reasons why you wanna name light as `light_num` if `lightDef.name` isn't defined? I think user may want the empty strings name (default in Three.js) in that case.

Reply:
See https://github.com/KhronosGroup/glTF-Validator/issues/93 — I'm increasingly thinking we should ensure that all nodes have unique names, even if it means changing user-specified names. Duplicate names are a common source of animation bugs. Because empty strings are practically guaranteed to be non-unique, I don't think we should be trying to preserve those.

Reply:
I'm one of the users who don't want loader to pollute the names as default. And currently glTF spec allows non-unique name.

But I can understand non-unique name can sometimes cause problems so wondering if I can first add a method to `SceneUtils` or somewhere which ensures non-unique name in a scene and ask users to explicit call. We can move it into the loader later if glTF spec becomes to recommend unique name.

Let's make a thread and discuss unique-name handling in Three.js there when necessary. And I want glTF spec to update or add a node about unique name first.

Reply:
The validator thread (https://github.com/KhronosGroup/glTF-Validator/issues/93) goes into some detail on this — there are many tools that expect or require uniquely named objects, even though the glTF spec does not. Currently renaming seems to be the recommended way to handle this. But I would be OK with consolidating the renaming logic into one place rather than having one-offs like this, whether that's within GLTFLoader somewhere or in a Utils class.

Technically we don't need globally unique names — just unique within the asset, so that an AnimationMixer has references to nodes that are both (a) unique and (b) portable if animations are loaded from a different file than the mesh.

Reply:
> there are many tools that expect or require uniquely named objects, even though the glTF spec does not.

I think first glTF spec should add a note, it'd be a lack of consideration for asset runs anywhere.

Reply:
You may want to comment on the thread, then. The current consensus is that the spec will not require unique names.

Reply:
I won't because I don't request unique-name in the spec.